src/components/AggregationPanel/AggregationPanelComponents.js

```diff
@@ -95,3 +97,116 @@ export const ButtonElement = ({ item, showNote }) => {
     </div>
   );
 };
+
+export const PanelElement = ({ item, showNote, objectId, depth = 0 }) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [panelData, setPanelData] = useState(null);
+
+  const fetchPanelData = async () => {
```

Comment:
How about wrapping `fetchPanelData`, `handleToggle` and `handleRefresh` functions into `useCallback`?